Create gw_docker_build_docs

mmann1123 opened the pull request:
Allows us to build docs on the fly, one command

jgrss commented on dockerfiles/gw_docker_build_docs:
```diff
@@ -0,0 +1,29 @@
+    pip \
+    g++ 
+
+RUN pip install Cython numpy  
```
Do we want/need to pin the versions on anything here?

mmann1123 replied:
This build seems to be working. But I differ to you, I don't have a clear sense of how best to handle this.

jgrss replied:
It might be a good idea for the Docker files.

jgrss commented on dockerfiles/gw_docker_build_docs:
```diff
@@ -0,0 +1,29 @@
+RUN pip install sklearn-xarray  pip-tools rtree ipykernel sphinx graphviz numpydoc sphinx_automodapi sphinxcontrib.bibtex==1.0
+
+RUN python -c "import geowombat as gw"
+RUN python -c "import rasterio;from rasterio.crs import CRS; cc = CRS.from_epsg(4326)"
```
These are just tests, right?

mmann1123 replied:
Yes that's just a basic test to make sure things are set up.

mmann1123 commented:
Ok I just added the versions and added jax & jaxlib. I was able to build the image and generate docs.  

Not sure how to interpret the circle i build error

Should I have it automatically branch, commit and create a pull request?

jgrss commented:
> Ok I just added the versions and added jax & jaxlib. I was able to build the image and generate docs.
> 
> Not sure how to interpret the circle i build error
> 
> Should I have it automatically branch, commit and create a pull request?

I think you need to merge `main` into this branch because this branch doesn't have `.circleci`.

jgrss commented:
> > Ok I just added the versions and added jax & jaxlib. I was able to build the image and generate docs.
> > Not sure how to interpret the circle i build error
> > Should I have it automatically branch, commit and create a pull request?
> 
> I think you need to merge `main` into this branch because this branch doesn't have `.circleci`.

So I think just do:
```commandline
git checkout main
git pull origin main
git checkout mmann1123-build-docs
git merge main
git push origin mmann1123-build-docs
```

jgrss reviewed (approved):
Looks good. I merged the latest from `main` into this branch. It should be ready to merge whenever.
